test/graphql/CompanyUser/Mutations/updateMyForgottenPassword.test.ts

```diff
@@ -0,0 +1,204 @@
+    const { user } = await TestClientGenerator.company();
+    const variables = await generateVariables({ user, newPassword: "does_not_have_upper_case" });
+    const apolloClient = client.loggedOut();
+    const { errors } = await performQuery(apolloClient, variables);
+    expect(errors).toEqualGraphQLErrorType("PasswordWithoutUppercaseError");
+  });
+
+  it("returns an error if the password does not have a lower case character", async () => {
+    const { user } = await TestClientGenerator.company();
+    const variables = await generateVariables({ user, newPassword: "DOES_NOT_HAVE_LOWER_CASE" });
+    const apolloClient = client.loggedOut();
+    const { errors } = await performQuery(apolloClient, variables);
+    expect(errors).toEqualGraphQLErrorType("PasswordWithoutLowercaseError");
+  });
+
+  it("returns an error if the password has spaces", async () => {
+    const { user } = await TestClientGenerator.company();
+    const variables = await generateVariables({ user, newPassword: "password With Spaces 1" });
+    const apolloClient = client.loggedOut();
+    const { errors } = await performQuery(apolloClient, variables);
+    expect(errors).toEqualGraphQLErrorType("PasswordWithSpacesError");
+  });
+
+  it("returns an error if the token expires", async () => {
+    const { user } = await TestClientGenerator.company();
+    const apolloClient = client.loggedOut();
+    const variables = await generateVariables({ user, expiresIn: "0d" });
+    const { errors } = await performQuery(apolloClient, variables);
+    expect(errors).toEqualGraphQLErrorType(UnauthorizedError.name);
+  });
+
+  it("returns an error if the token has an encoded applicant user", async () => {
+    const password = "SecurePass100";
+    const { user } = await TestClientGenerator.applicant({ password });
+    const apolloClient = client.loggedOut();
+    const variables = await generateVariables({ user });
+    const { errors } = await performQuery(apolloClient, variables);
+    expect(errors).toEqualGraphQLErrorType(CompanyUserNotFoundError.name);
+    const persistedUser = await UserRepository.findByUuid(user.uuid!);
+    expect(persistedUser.credentials).toBeInstanceOf(FiubaCredentials);
+  });
+
+  it("returns an error if the token has an encoded admin user", async () => {
+    const password = "SecurePass100";
+    const { user } = await TestClientGenerator.admin({ secretary: Secretary.extension, password });
+    const apolloClient = client.loggedOut();
+    const variables = await generateVariables({ user });
+    const { errors } = await performQuery(apolloClient, variables);
+    expect(errors).toEqualGraphQLErrorType(CompanyUserNotFoundError.name);
+    const persistedUser = await UserRepository.findByUuid(user.uuid!);
+    expect(persistedUser.credentials).toBeInstanceOf(FiubaCredentials);
+  });
+
+  it("returns an error if there is a current user from a approved company", async () => {
```

Comment:
no entendi esta descripcion, creo que seria algo como 

"returns an error if an approved company user is already logged in"

Reply:
(idem los de abajo)